### PR TITLE
[catcollar] Initial LibOS Implementation

### PR DESCRIPTION
**Pull request by ppenna**
Description
========

In this PR I introduce the initial implementation of Catcollar, our LibOS over [liburing](https://github.com/axboe/liburing).

### CODE REVIEW

File: src/catcollar/runtime/scheduler.rs [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        WaitFuture,
+    },
+};
+use ::std::time::Instant;
+
+//==============================================================================
+// Trait Implementations
+//==============================================================================
+
+/// Scheduler Runtime Trait Implementation for I/O User Ring Runtime
```
Comment (BrianZill) [inline src/catcollar/runtime/scheduler.rs:26]: Missing period.  :-)

File: src/catcollar/runtime/utils.rs [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    distributions::Standard,
+    prelude::Distribution,
+};
+use ::runtime::utils::UtilsRuntime;
+
+//==============================================================================
+// Trait Implementations
+//==============================================================================
+
+/// Utilities Runtime Trait Implementation for I/O User Ring Runtime
```
Comment (BrianZill) [inline src/catcollar/runtime/utils.rs:19]: Another missing period.

**Review by BrianZill — APPROVED** [review]
Looks good, except for those two comments.

File: src/catcollar/runtime/scheduler.rs (hunk 1 above)
Comment (ppenna) [reply src/catcollar/runtime/scheduler.rs:26]: Sure.

File: src/catcollar/runtime/utils.rs (hunk 2 above)
Comment (ppenna) [reply src/catcollar/runtime/utils.rs]: Sure.

**Pull request merged**